src/Psy/Plugin/AbstractPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Psy\Plugin;
+
+abstract class AbstractPlugin
+{
+    public static function register()
+    {
+        Manager::register(new static(), static::getName());
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getName()
+    {
+        $class = new \ReflectionClass(get_called_class());
+
+        return preg_replace('#Plugin$#', '', $class->getShortName());
```

Comment:
But what if there are two people with `DevToolsPlugin` or `AwesomePlugin`? Is disambiguating by the first part of the short class name the best way to go?